matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncResponseHandler.kt

```diff
@@ -50,7 +51,7 @@ private const val GET_GROUP_DATA_WORKER = "GET_GROUP_DATA_WORKER"
 
 internal class SyncResponseHandler @Inject constructor(
         @SessionDatabase private val monarchy: Monarchy,
-        @SessionId private val sessionId: String,
+        private val session: Session,
```

Comment:
I wasn't sure about this change, technically this works as the `SyncThread` uses a `Dagger Provider` so the instance is always correct but it feels a little overkill~

Reply:
we need a session instance in order to call 

```kotlin
sessionListeners.dispatch(session) { session, listener ->
    listener.onNewInvitedRoom(session, roomId)
}
```

I could pass the `SessionManager` to the `SyncResponseHandler` like the `SessionListeners` was previously doing

Reply:
https://github.com/vector-im/element-android/pull/4480/commits/405cda7a2f2aaf4368a474a9a4620d893fe934f9 it's an inlined version of what the `SessionListeners` was doing

Reply:
I think we could also replace onNewInvitedRoom by listening to the Flow<SyncResponse> instead

Reply:
@ganfra yes, please do that in another PR :p